pkg/services/ngalert/accesscontrol/rules.go

```diff
@@ -113,14 +139,16 @@ func (r *RuleService) AuthorizeAccessToRuleGroup(ctx context.Context, user ident
 func (r *RuleService) AuthorizeRuleChanges(ctx context.Context, user identity.Requester, change *store.GroupDelta) error {
 	namespaceScope := dashboards.ScopeFoldersProvider.GetResourceScopeUID(change.GroupKey.NamespaceUID)
 
-	rules, ok := change.AffectedGroups[change.GroupKey]
-	if ok { // not ok can be when user creates a new rule group or moves existing alerts to a new group
+	rules, existingGroup := change.AffectedGroups[change.GroupKey]
+	if existingGroup { // not existingGroup can be when user creates a new rule group or moves existing alerts to a new group
 		if err := r.AuthorizeAccessToRuleGroup(ctx, user, rules); err != nil { // if user is not authorized to do operation in the group that is being changed
 			return err
 		}
-	} else if len(change.Delete) > 0 {
-		// add a safeguard in the case of inconsistency. If user hit this then there is a bug in the calculating of changes struct
-		return fmt.Errorf("failed to authorize changes in rule group %s. Detected %d deletes but group was not provided", change.GroupKey.RuleGroup, len(change.Delete))
+	} else {
+		if len(change.Delete) > 0 {
+			// add a safeguard in the case of inconsistency. If user hit this then there is a bug in the calculating of changes struct
+			return fmt.Errorf("failed to authorize changes in rule group %s. Detected %d deletes but group was not provided", change.GroupKey.RuleGroup, len(change.Delete))
+		}
 	}
```

Comment:
This should remove the diff for these lines, logic is now the same as it was before.

```suggestion
	} else if len(change.Delete) > 0 {
		// add a safeguard in the case of inconsistency. If user hit this then there is a bug in the calculating of changes struct
		return fmt.Errorf("failed to authorize changes in rule group %s. Detected %d deletes but group was not provided", change.GroupKey.RuleGroup, len(change.Delete))
	}
```